### PR TITLE
bumps outdated deps versions to be current + drop support for java 8

### LINKED ISSUE #22401: Environment variable with whitespace as the name crashes metabase 0.43.0
**Describe the bug**
If an environment variable with just spaces for the name exists, metabase 0.43.0 crashes during startup.  0.42.X didn't crash.

**Logs**
Here are logs from running the startup script:
```
bash-5.1$ env " "=test app/run_metabase.sh
Picked up JAVA_TOOL_OPTIONS: -Xmx2g
Warning: environ value jdk-11.0.15+10 for key :java-version has been overwritten with 11.0.15
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
2022-05-03 22:59:04,887 INFO metabase.util :: Maximum memory available to JVM: 1.9 GB
2022-05-03 22:59:06,712 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
 For more information, see https://metabase.com/docs/latest/operations-guide/encrypting-database-details-at-rest.html
2022-05-03 22:59:09,307 INFO driver.impl :: Registered abstract driver :sql  🚚
2022-05-03 22:59:09,310 INFO metabase.util ::  ⮦ Load driver :sql took 83.4 ms
2022-05-03 22:59:09,315 INFO driver.impl :: Registered abstract driver :sql-jdbc (parents: [:sql]) 🚚
2022-05-03 22:59:09,317 INFO metabase.util :: Load driver :sql-jdbc took 92.8 ms
2022-05-03 22:59:09,317 INFO driver.impl :: Registered driver :h2 (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:09,330 INFO driver.impl :: Registered driver :mysql (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:09,340 INFO driver.impl :: Registered driver :postgres (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:10,506 INFO metabase.core ::
Metabase v0.43.0 (ee686fc release-x.43.x)

Copyright © 2022 Metabase, Inc.

Metabase Enterprise Edition extensions are NOT PRESENT.
2022-05-03 22:59:10,514 INFO metabase.core :: Starting Metabase in STANDALONE mode
2022-05-03 22:59:10,542 INFO metabase.server :: Launching Embedded Jetty Webserver with config:
 {:port 3000, :host "0.0.0.0"}

2022-05-03 22:59:10,565 INFO metabase.core :: Starting Metabase version v0.43.0 (ee686fc release-x.43.x) ...
2022-05-03 22:59:10,581 INFO metabase.core :: System info:
 {"file.encoding" "UTF-8",
 "java.runtime.name" "OpenJDK Runtime Environment",
 "java.runtime.version" "11.0.15+10",
 "java.vendor" "Eclipse Adoptium",
 "java.vendor.url" "https://adoptium.net/",
 "java.version" "11.0.15",
 "java.vm.name" "OpenJDK 64-Bit Server VM",
 "java.vm.version" "11.0.15+10",
 "os.name" "Linux",
 "os.version" "4.14.273-207.502.amzn2.x86_64",
 "user.language" "en",
 "user.timezone" "GMT"}

2022-05-03 22:59:10,582 INFO metabase.plugins :: Loading plugins in /plugins...
2022-05-03 22:59:10,677 INFO util.files :: Extract file /modules/sparksql.metabase-driver.jar -> /plugins/sparksql.metabase-driver.jar
2022-05-03 22:59:10,717 INFO util.files :: Extract file /modules/sqlserver.metabase-driver.jar -> /plugins/sqlserver.metabase-driver.jar
2022-05-03 22:59:10,724 INFO util.files :: Extract file /modules/presto-jdbc.metabase-driver.jar -> /plugins/presto-jdbc.metabase-driver.jar
2022-05-03 22:59:10,765 INFO util.files :: Extract file /modules/sqlite.metabase-driver.jar -> /plugins/sqlite.metabase-driver.jar
2022-05-03 22:59:10,779 INFO util.files :: Extract file /modules/vertica.metabase-driver.jar -> /plugins/vertica.metabase-driver.jar
2022-05-03 22:59:10,779 INFO util.files :: Extract file /modules/redshift.metabase-driver.jar -> /plugins/redshift.metabase-driver.jar
2022-05-03 22:59:10,784 INFO util.files :: Extract file /modules/presto.metabase-driver.jar -> /plugins/presto.metabase-driver.jar
2022-05-03 22:59:10,785 INFO util.files :: Extract file /modules/druid.metabase-driver.jar -> /plugins/druid.metabase-driver.jar
2022-05-03 22:59:10,788 INFO util.files :: Extract file /modules/mongo.metabase-driver.jar -> /plugins/mongo.metabase-driver.jar
2022-05-03 22:59:10,800 INFO util.files :: Extract file /modules/oracle.metabase-driver.jar -> /plugins/oracle.metabase-driver.jar
2022-05-03 22:59:10,801 INFO util.files :: Extract file /modules/snowflake.metabase-driver.jar -> /plugins/snowflake.metabase-driver.jar
2022-05-03 22:59:10,918 INFO util.files :: Extract file /modules/googleanalytics.metabase-driver.jar -> /plugins/googleanalytics.metabase-driver.jar
2022-05-03 22:59:10,927 INFO util.files :: Extract file /modules/bigquery-cloud-sdk.metabase-driver.jar -> /plugins/bigquery-cloud-sdk.metabase-driver.jar
2022-05-03 22:59:10,960 INFO util.files :: Extract file /modules/presto-common.metabase-driver.jar -> /plugins/presto-common.metabase-driver.jar
2022-05-03 22:59:11,050 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :hive-like...
2022-05-03 22:59:11,057 INFO driver.impl :: Registered abstract driver :hive-like (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:11,057 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sparksql...
2022-05-03 22:59:11,058 INFO driver.impl :: Registered driver :sparksql (parents: [:hive-like]) 🚚
2022-05-03 22:59:11,063 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlserver...
2022-05-03 22:59:11,064 INFO driver.impl :: Registered driver :sqlserver (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:11,077 INFO plugins.dependencies :: Plugin 'Metabase Presto JDBC Driver' depends on plugin 'Presto Common Driver'
2022-05-03 22:59:11,078 INFO plugins.dependencies :: Metabase Presto JDBC Driver dependency {:plugin Presto Common Driver} satisfied? false
2022-05-03 22:59:11,079 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Presto JDBC Driver"]
2022-05-03 22:59:11,081 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :sqlite...
2022-05-03 22:59:11,082 INFO driver.impl :: Registered driver :sqlite (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:11,085 INFO plugins.dependencies :: Metabase cannot initialize plugin Metabase Vertica Driver due to required dependencies. Metabase requires the Vertica JDBC driver in order to connect to Vertica databases, but we can't ship it as part of Metabase due to licensing restrictions. See https://metabase.com/docs/latest/administration-guide/databases/vertica.html for more details.

2022-05-03 22:59:11,086 INFO plugins.dependencies :: Metabase Vertica Driver dependency {:class com.vertica.jdbc.Driver} satisfied? false
2022-05-03 22:59:11,086 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase Presto JDBC Driver"]
2022-05-03 22:59:11,091 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :redshift...
2022-05-03 22:59:11,091 INFO driver.impl :: Registered driver :redshift (parents: [:postgres]) 🚚
2022-05-03 22:59:11,093 INFO plugins.dependencies :: Plugin 'Metabase Presto Driver' depends on plugin 'Presto Common Driver'
2022-05-03 22:59:11,093 INFO plugins.dependencies :: Metabase Presto Driver dependency {:plugin Presto Common Driver} satisfied? false
2022-05-03 22:59:11,094 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase Presto Driver" "Metabase Presto JDBC Driver"]
2022-05-03 22:59:11,095 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :druid...
2022-05-03 22:59:11,096 INFO driver.impl :: Registered driver :druid  🚚
2022-05-03 22:59:11,099 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :mongo...
2022-05-03 22:59:11,099 INFO driver.impl :: Registered driver :mongo  🚚
2022-05-03 22:59:11,102 INFO plugins.dependencies :: Metabase cannot initialize plugin Metabase Oracle Driver due to required dependencies. Metabase requires the Oracle JDBC driver in order to connect to Oracle databases, but we can't ship it as part of Metabase due to licensing restrictions. See https://metabase.com/docs/latest/administration-guide/databases/oracle.html for more details.

2022-05-03 22:59:11,102 INFO plugins.dependencies :: Metabase Oracle Driver dependency {:class oracle.jdbc.OracleDriver} satisfied? false
2022-05-03 22:59:11,103 INFO plugins.dependencies :: Plugins with unsatisfied deps: ["Metabase Vertica Driver" "Metabase Presto Driver" "Metabase Oracle Driver" "Metabase Presto JDBC Driver"]
2022-05-03 22:59:11,116 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :snowflake...
2022-05-03 22:59:11,116 INFO driver.impl :: Registered driver :snowflake (parents: [:sql-jdbc]) 🚚
2022-05-03 22:59:11,118 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :googleanalytics...
2022-05-03 22:59:11,119 INFO driver.impl :: Registered driver :googleanalytics  🚚
2022-05-03 22:59:11,123 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :bigquery-cloud-sdk...
2022-05-03 22:59:11,123 INFO driver.impl :: Registered driver :bigquery-cloud-sdk (parents: [:sql]) 🚚
2022-05-03 22:59:11,125 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :presto-common...
2022-05-03 22:59:11,125 INFO driver.impl :: Registered abstract driver :presto-common  🚚
2022-05-03 22:59:11,126 INFO plugins.dependencies :: Metabase Presto Driver dependency {:plugin Presto Common Driver} satisfied? true
2022-05-03 22:59:11,126 INFO plugins.dependencies :: Metabase Presto JDBC Driver dependency {:plugin Presto Common Driver} satisfied? true
2022-05-03 22:59:11,127 DEBUG plugins.initialize :: Dependencies satisfied; these plugins will now be loaded: ["Metabase Presto JDBC Driver" "Metabase Presto Driver"]
2022-05-03 22:59:11,127 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :presto-jdbc...
2022-05-03 22:59:11,127 INFO driver.impl :: Registered driver :presto-jdbc (parents: [:presto-common]) 🚚
2022-05-03 22:59:11,128 DEBUG plugins.lazy-loaded-driver :: Registering lazy loading driver :presto...
2022-05-03 22:59:11,128 INFO driver.impl :: Registered driver :presto (parents: [:presto-common]) 🚚
2022-05-03 22:59:11,129 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
2022-05-03 22:59:11,131 INFO db.setup :: Verifying postgres Database Connection ...
2022-05-03 22:59:11,395 INFO db.setup :: Successfully verified PostgreSQL 12.8 application database connection. ✅
2022-05-03 22:59:11,396 INFO db.setup :: Running Database Migrations...
2022-05-03 22:59:11,397 INFO db.setup :: Setting up Liquibase...
2022-05-03 22:59:11,517 ERROR metabase.core :: Metabase Initialization FAILED
java.lang.NullPointerException
	at liquibase.configuration.AbstractMapConfigurationValueProvider.keyMatches(AbstractMapConfigurationValueProvider.java:140)
	at liquibase.configuration.pro.EnvironmentValueProvider.keyMatches(Unknown Source)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.lookupProvidedValue(AbstractMapConfigurationValueProvider.java:105)
	at liquibase.configuration.pro.EnvironmentValueProvider.lookupProvidedValue(Unknown Source)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:52)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:123)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:93)
	at liquibase.license.pro.DaticalTrueLicenseService.licenseIsInstalled(Unknown Source)
	at liquibase.license.pro.DaticalTrueLicenseService.licenseIsValid(Unknown Source)
	at liquibase.configuration.pro.EnvironmentValueProvider.lookupProvidedValue(Unknown Source)
	at liquibase.configuration.AbstractMapConfigurationValueProvider.getProvidedValue(AbstractMapConfigurationValueProvider.java:52)
	at liquibase.configuration.LiquibaseConfiguration.getCurrentConfiguredValue(LiquibaseConfiguration.java:123)
	at liquibase.configuration.ConfigurationDefinition.getCurrentConfiguredValue(ConfigurationDefinition.java:93)
	at liquibase.configuration.ConfigurationDefinition.getCurrentValue(ConfigurationDefinition.java:61)
	at liquibase.database.AbstractJdbcDatabase.getDatabaseChangeLogLockTableName(AbstractJdbcDatabase.java:641)
	at liquibase.database.core.PostgresDatabase.getDatabaseChangeLogLockTableName(PostgresDatabase.java:162)
	at liquibase.changelog.ChangeLogParameters.<init>(ChangeLogParameters.java:45)
	at liquibase.Liquibase.<init>(Liquibase.java:119)
	at metabase.db.liquibase$liquibase.invokeStatic(liquibase.clj:48)
	at metabase.db.liquibase$liquibase.invoke(liquibase.clj:48)
	at metabase.db.liquibase$fn__30484$do_with_liquibase__30489$fn__30490.invoke(liquibase.clj:59)
	at metabase.db.liquibase$fn__30484$do_with_liquibase__30489.invoke(liquibase.clj:51)
	at metabase.db.setup$fn__34820$migrate_BANG___34825$fn__34826.invoke(setup.clj:63)
	at metabase.db.setup$fn__34820$migrate_BANG___34825.invoke(setup.clj:40)
	at metabase.db.setup$fn__34879$run_schema_migrations_BANG___34884$fn__34885.invoke(setup.clj:121)
	at metabase.db.setup$fn__34879$run_schema_migrations_BANG___34884.invoke(setup.clj:115)
	at metabase.db.setup$fn__34931$setup_db_BANG___34936$fn__34937$fn__34940$fn__34941.invoke(setup.clj:147)
	at metabase.util$do_with_us_locale.invokeStatic(util.clj:715)
	at metabase.util$do_with_us_locale.invoke(util.clj:701)
	at metabase.db.setup$fn__34931$setup_db_BANG___34936$fn__34937$fn__34940.invoke(setup.clj:145)
	at metabase.db.setup$fn__34931$setup_db_BANG___34936$fn__34937.invoke(setup.clj:144)
	at metabase.db.setup$fn__34931$setup_db_BANG___34936.invoke(setup.clj:138)
	at metabase.db$setup_db_BANG_$fn__34966.invoke(db.clj:65)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:60)
	at metabase.db$setup_db_BANG_.invoke(db.clj:51)
	at metabase.core$init_BANG_.invokeStatic(core.clj:96)
	at metabase.core$init_BANG_.invoke(core.clj:79)
	at metabase.core$start_normally.invokeStatic(core.clj:137)
	at metabase.core$start_normally.invoke(core.clj:131)
	at metabase.core$_main.invokeStatic(core.clj:170)
	at metabase.core$_main.doInvoke(core.clj:164)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
2022-05-03 22:59:11,523 INFO metabase.core :: Metabase Shutting Down ...
2022-05-03 22:59:11,523 INFO metabase.server :: Shutting Down Embedded Jetty Webserver
2022-05-03 22:59:11,528 INFO metabase.core :: Metabase Shutdown COMPLETE
```

**To Reproduce**
1. Create an environment variable with a name of " " and attempt to start metabase. `env " "=test run_metabase.sh` reproduces it
2. See error during startup

**Expected behavior**
Metabase does not crash

**Information about your Metabase Installation:**- Metabase version: (e.g., 0.43.0)
- Metabase hosting environment: docker on AWS ECS
- Metabase internal database: Postgres

**Severity**
Low.  I have a workaround - `env -u` can remove the environment variable and the environment variable shouldn't exist in the first place.

**Additional context**
This looks like a liquibase bug that has a fix committed already: https://github.com/liquibase/liquibase/issues/2664

### PR DESCRIPTION
**Pull request by escherize**
The usual deps bump for a new major version.

### CODE REVIEW

**Comment by rlotun** [comment]
does it make sense to do the liquidbase update in a separate PR that can be backported since https://github.com/metabase/metabase/issues/22401 is a regression (that is spaces in env vars *used* to work in previous Metabase versions)

**Review by dpsutton — APPROVED** [review]

**Comment by qnkhuat** [comment]
oh we're bumping h2 too? I think we're doing it separately in https://github.com/metabase/metabase/pull/22354

**Review by flamber — DISMISSED** [review]
Only upgrade Liquibase in this PR. Nothing else.
You cannot just bump core stuff like Jetty and H2 without everything exploding.

File: deps.edn [hunk 1]
```diff
@@ -61,19 +61,19 @@
...
   io.forward/yaml                           {:mvn/version "1.0.11"              ; Clojure wrapper for YAML library SnakeYAML. Don't upgrade yet, new version doesn't support Java 8 (see https://github.com/owainlewis/yaml/issues/37)
                                              :exclusions  [org.clojure/clojure
                                                            org.flatland/ordered
                                                            org.yaml/snakeyaml]}
   io.github.resilience4j/resilience4j-retry {:mvn/version "1.7.1"}              ; Support for retrying operations
   javax.xml.bind/jaxb-api                   {:mvn/version "2.4.0-b180830.0359"} ; add the `javax.xml.bind` classes which we're still using but were removed in Java 11
-  joda-time/joda-time                       {:mvn/version "2.10.13"}
-  kixi/stats                                {:mvn/version "0.4.4"               ; Various statistic measures implemented as transducers
+  joda-time/joda-time                       {:mvn/version "2.10.14"}
+  kixi/stats                                {:mvn/version "0.5.5"               ; Various statistic measures implemented as transducers
```
Comment (camsaul) [inline deps.edn:72]: I think this might have broken something last time I tried it. Just a heads up in case tests fail on X-Rays

**Review by camsaul — APPROVED** [review]
It seems like one of the version bumps dropped support for Java 8. Since we want to drop support for Java 8 in 44 anyway you can probably just go ahead and remove the Java 8 nodes from CI I think

**Comment by escherize** [comment]
I'll keep messing with this until the tests are passing. Obviously we should upgrade as many libraries as we can.

**Comment by diogormendes** [comment]
I think we can also go and remove 8 from Github like in here: https://github.com/metabase/metabase/blob/75747c1931f68d11feeb45a2aabc2e7a723af8f7/.github/workflows/uberjar.yml#L47

**Review by dpsutton — APPROVED** [review]
Looks good. The tests seem to be failing because we are now getting records instead of maps with revisions. That's really weird.

**Pull request merged**